SerialPrograms/Source/PokemonSV/Programs/AutoStory/PokemonSV_MenuOption.cpp

```diff
@@ -0,0 +1,214 @@
+        throw OperationFailedException(
+            ErrorReport::SEND_ERROR_REPORT, m_console,
+            "MenuOption::get_selected_index(): Invalid cursor slot."
+        );
+    }
+    // cout << "selected index:" + std::to_string(selected_index) << endl;
+
+    return selected_index;
+}
+
+std::string MenuOption::read_option(const ImageViewRGB32& cropped) const{
+    // cropped.save("test.png");
+    const auto ocr_result = MenuOptionReader::instance().read_substring(
+        m_console, m_language, 
+        cropped, OCR::BLACK_TEXT_FILTERS()
+    );
+
+    std::multimap<double, OCR::StringMatchData> results;
+    if (!ocr_result.results.empty()){
+        for (const auto& result : ocr_result.results){
+            results.emplace(result.first, result.second);
+        }
+    }
+
+    if (results.empty()){
+        if (m_language == Language::German){
+            // German Menu Option uses numbers for text speed settings, which string OCR has trouble with detecting.
+            int16_t number = read_number(m_console, cropped);
+            switch (number){
+                case 1:
+                    return "slow";
+                case 2:
+                    return "normal";
+                case 3:
+                    return "fast";
+            }
+        }
+        return "";
+    }
+
+    if (results.size() > 1){
+        throw OperationFailedException(
+            ErrorReport::SEND_ERROR_REPORT, 
+            m_console,
+            "MenuOption::read_option(): Unable to read item. Ambiguous or multiple results."
+        );
+    }
+
+    return results.begin()->second.token;
+}
+
+int16_t MenuOption::read_number(
+    Logger& logger, 
+    const ImageViewRGB32& cropped
+) const{
+
+    int16_t number = (int16_t)OCR::read_number_waterfill(logger, cropped, 0xff000000, 0xff808080);
+
+    if (number < 1 || number > 3){
+        number = -1; 
+    }
+    return (int16_t)number;
+}
+
+std::string MenuOption::read_option_item() const{
+    m_context.wait_for_all_requests();
+    VideoSnapshot screen = m_console.video().snapshot();
+    
+    ImageFloatBox selected_box = m_boxes_item[get_selected_index(screen)];
+    ImageViewRGB32 cropped = extract_box_reference(screen, selected_box);
+    return read_option(cropped);
+
+}
+
+std::string MenuOption::read_option_toggle() const{
+    m_context.wait_for_all_requests();
+    VideoSnapshot screen = m_console.video().snapshot();
+
+    ImageFloatBox selected_box = m_boxes_toggle[get_selected_index(screen)];
+    ImageViewRGB32 cropped = extract_box_reference(screen, selected_box);
+
+    return read_option(cropped);
+}
+
+
+void MenuOption::move_to_option(const MenuOptionItemEnum target_option_enum) const{
+    std::string current_option_slug = read_option_item();
+    MenuOptionItem current_option = menu_option_item_lookup_by_slug(current_option_slug);
+
+    MenuOptionItem target_option = menu_option_item_lookup_by_enum(target_option_enum);
```

Comment:
Both of these can be const-ref instead of value to avoid copy.

Reply:
I can change target_option to const ref. But not current_option, since it gets reassigned in the loop.